tools/check.py

```diff
@@ -667,6 +667,45 @@ def command_line():
 
     return parser.parse_args()
 
+def check_required_files(dir_to_validate):
+    """Check if required files exists."""
+    REQUIRED_FILES = ["01-*.md",
+                      "discussion.md",
+                      "index.md",
+                      "instructors.md",
+                      "LICENSE.md",
+                      "motivation.md",
+                      "README.md",
+                      "reference.md"]
+    valid = True
+
+    for required in REQUIRED_FILES:
+        req_fn = os.path.join(dir_to_validate, required)
+        if not glob.glob(req_fn):
+            logging.error(
+                "Missing file {0}.".format(required))
+            valid = False
+
+    return valid
+
+def get_files_to_validate(file_or_path):
+    """Generate list of files to validate."""
+    files_to_validate = []
+    dir_to_validate = None
+
+    for fn in file_or_path:
+        if os.path.isdir(fn):
+            search_str = os.path.join(fn, "*.md")
+            files_to_validate.extend(glob.glob(search_str))
+            dir_to_validate = fn
```

Comment:
The user can specify any number of directories at the command line (`nargs=*`). But as written, this would only return `dir_to_validate` for the last directory specified.

`dir_to_validate` should be a list, so line 724 can just be a for loop. (which is skipped when the list is empty)

The multi-directory support is useful for, eg, running the validator on an entire collection of lessons in one go. (I use the feature for QA) Otherwise this is looking really close.